framework/src/samplers/Sampler.C

```diff
@@ -266,6 +267,15 @@ Sampler::getRand(const unsigned int index)
   return _generator.rand(index);
 }
 
+uint32_t
+Sampler::getRandl(unsigned int index, uint32_t lower, uint32_t upper)
+{
+  mooseAssert(index < _generator.size(), "The seed number index does not exists.");
+  TIME_SECTION(_perf_get_randl);
+
+  return _generator.randl(index, lower, upper);
```

Comment:
I bet this that this is a little too small of a scope to time. I would recommend that after you look at this a bit that you take it back out. The timing code has got to be on the same order of the random generation code.

Reply:
I will take those out, I was trying to figure out some timing issue last week when I added these.